dom.d.ts

```diff
@@ -0,0 +1,44 @@
+type Nil = null | undefined;
+type Primitive = string | number | bigint | boolean;
+
+type Prop = Nil | Primitive;
+type Props = Nil | Record<string, Prop>;
+
+type Child = Nil | Primitive | Node | Child[];
+
+export function E<N extends HTMLInputElement>(name: 'input', props?: Props, ...children: Child[]): N;
+export function E<N extends HTMLElement>(name: string, props?: Props, ...children: Child[]): N;
+
+export function S(name: string, props?: Props, ...children: Child[]): SVGElement;
+
+export function F(...children: Child[]): Node;
```

Comment:
This always returns a `DocumentFragment`. (Not sure if TS has that pre-defined, though.)